BM25.ts

```diff
@@ -59,10 +59,15 @@ export default function BM25(
   );
   const averageDocumentLength =
     documentLengths.reduce((a, b) => a + b, 0) / documents.length;
+  const idfByKeyword = keywords.reduce((obj, keyword) => {
+    obj[keyword] = getIDF(keyword, documents);
+    return obj;
+  }, {});
+
```

Comment:
Let's pass in a Map object instead of an arbitrary Object. This way we can type check for indexes that result in undefined values.

For example

  const idfByKeyword = keywords.reduce((idfMap, keyword) => {
    idfMap.set(keyword, getIDF(keyword, documents));
    return idfMap;
  }, new Map<type, type>());

Reply:
Cool, done.